seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/SourceSplitEnumeratorTask.java

```diff
@@ -141,7 +141,7 @@ public void triggerBarrier(Barrier barrier) throws Exception {
         final long barrierId = barrier.getId();
         Serializable snapshotState = null;
         byte[] serialize = null;
-        synchronized (enumeratorContext) {
+        synchronized (source) {
```

Comment:
@ashulin  @Hisoka-X @hailin0 PTAL

Reply:
This is a warning, because the lock object is not final. So I chose a final modified lock.

Reply:
> Useing private object field as lock object

Done.

Reply:
> > Useing private object field as lock object
> 
> Done.

The properties of `SourceSplitEnumeratorTask` need to be serialized, so I removed the lock.